compiler/rustc_codegen_llvm/src/back/lto.rs

```diff
@@ -313,7 +313,9 @@ fn fat_lto(
         for (bc_decoded, name) in serialized_modules {
             let _timer = cgcx
                 .prof
-                .generic_activity_with_arg("LLVM_fat_lto_link_module", format!("{:?}", name));
+                .generic_activity_with_arg_recorder("LLVM_fat_lto_link_module", |recorder| {
+                    recorder.record_arg(format!("{:?}", name))
```

Comment:
These changes are fine but looking at this more closely, does `name` actually need to be a `CString` at all?

Reply:
I think it's because the name is [computed once in preparation](https://github.com/rust-lang/rust/blob/ed6c958ee4bf081deec951996ace9c508360c1d9/compiler/rustc_codegen_llvm/src/back/lto.rs#L78), and then used in different places in fat and thin lto: 
- concatenated with other modules for different purposes (cached modules, serialized modules, green modules, etc)
- or as e.g. the module identifier for LLVM functions and data structures, or in error handling (where `CString` also feels out-of-place) and debug logging like here. 

I suppose this was done to avoid many different back-and-forth conversions (and/or done at a time before `rustc_codegen_ssa` had a `ModuleCodegen` with a String name), but looking around there's quite a few conversions  present still: a bit of cleanup could be interesting in this area.

I may look at that in another PR if that's OK ?